src/main/java/marquez/service/models/Source.java

```diff
@@ -35,4 +37,17 @@
   public Optional<String> getDescription() {
     return Optional.ofNullable(description);
   }
+
+  public boolean canAdd(@NonNull DatasetType datasetType) {
+    switch (qualifier) {
+      case DB:
+        return datasetType == DatasetType.DB_TABLE;
+      case STREAM:
+        return datasetType == DatasetType.STREAM;
+      case FILESYSTEM:
+        return datasetType == DatasetType.OBJECT;
+      default:
+        throw new AssertionError();
+    }
```

Comment:
this becomes unnecessary if we keep only one of those two enums. (side note, you can add fields and methods to enums so that you don't need a switch statement)